### PR TITLE
Convert a couple more ArgumentFactory implementations to use AbstractArgumentFactory

### PR DESCRIPTION
**Pull request by stevenschlansker**
based on feedback on #698 by @pennello

### CODE REVIEW

File: postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java [hunk 1]
```diff
@@ -13,15 +13,14 @@
...
+
 import org.jdbi.v3.core.argument.AbstractArgumentFactory;
 import org.jdbi.v3.core.argument.Argument;
-import org.jdbi.v3.core.argument.NullArgument;
 import org.jdbi.v3.core.config.ConfigRegistry;
 import org.postgresql.util.PGInterval;
 
-import java.sql.Types;
-import java.time.Duration;
-
```
Comment (pennello) [inline postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java]: This project should standardize on an import order.  ; ]

Comment (stevenschlansker) [inline postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java]: Yeah, we should...

**Comment by stevenschlansker** [comment]
It didn't look like it was as amenable to being converted.  In particular it doesn't get rid of the bulk of the work, which is selecting the correct `java.sql.Type` to use.

**Comment by pennello** [comment]
@stevenschlansker 
> It didn't look like it was as amenable to being converted. In particular it doesn't get rid of the bulk of the work, which is selecting the correct java.sql.Type to use.

Got it.  Then, perhaps it's worth a comment there to guard against future would-be developers following its pattern without really understanding the implementation differences?

File: postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java (hunk 1 above)
Comment (qualidafial) [inline postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java]: wut

**Pull request merged**